build-tools/packages/build-cli/src/library/repoPolicyCheck/fluidBuildTasks.ts

```diff
@@ -301,30 +231,64 @@ function eslintGetScriptDependencies(
 		throw new Error(`Unable to load eslint config file ${eslintConfig}. ${error}`);
 	}
 
-	let projects: string | string[] | undefined = config.parserOptions?.project;
-	if (projects === undefined) {
-		// If we don't have projects, our task needs to have dependent build scripts
-		return getDefaultTscTaskDependencies(root, json);
+	let projects = config.parserOptions?.project;
+	if (!Array.isArray(projects) && typeof projects !== "string") {
+		// config does not parse extends. So, "project" is what is set in top file.
+		if (projects === false || projects === null) {
+			// type based linting is disabled - assume no task prerequisites
+			return [];
+		}
+		// @typescript-eslint/parser allows true to mean use closest tsconfig.json, but we want
+		// explicit listings for dependency clarity.
+		if (projects === true) {
+			throw new Error(
+				`${json.name} eslint config 'project' is true, but should be list of tsconfig files.`,
```

Comment:
The appropriate action item(s) for a developer encountering this might not be clear. I wonder if something like the following would be more clear / actionable:
```suggestion
				`${json.name} eslint config's `parserOptions` setting has 'project' enabled, which is unsupported by fluid-build. Please specify one or more tsconfig files instead.`,
```

Reply:
Or will we hit this if `project` isn't specified at all as well?

Reply:
`undefined` is the final case within this block. Per Alex's suggestion there is now a better comment.